### ISSUE TITLE
Various issues on Raspberry Pi

### ISSUE THREAD

**Issue by johnstok** [Spotifyd/spotifyd#37]
Hi Simon,

I'd like to start with a big thanks for all the time you've invested in writing this. I've been trying it out on my RPi and run into a number of issues so I'd thought I'd list them out – I'm sure you're aware of many of them:

1. Skipping forward / back through a track doesn't seem to work correctly (it reports a `Vorbis error: InvalidSetup`);
2. Disconnecting and reconnecting the volume seems to be forgotten and defaults back to 100% – ouch;
3. When I play an album it seems to play one song and then stop;
4. Granularity of the volume control is very limited - my DAC has 64 vol positions but spotifyd doesn't seem to be that fine-grained;
5. Closing the Spotify app on iPhone and then restarting the app doesn't seem to detect that music is playing on spotfiyd;
6. Verbose output is too verbose – I tried to take look at the verbose logs but there was just so much output!

Happy to provide further info if I can :-)

**Comment by SimonPersson** [comment]
Hi!

Some of these issues are due to the underlying library, librespot. It's going through a bit of a rewrite, so I think there's no point in trying to solve them right now, better to wait and see how the rewrite works out. Anyway, I'll try to address some of these issues:

>Skipping forward / back through a track doesn't seem to work correctly (it reports a Vorbis error: InvalidSetup);

This could be a bug in librespot, Spotifyd, or perhaps even some issue with the cross compilation. Did you use the binary builds I've posted? If you have an x86 machine running Linux, could you try to run it there to and see if it works?

 > Disconnecting and reconnecting the volume seems to be forgotten and defaults back to 100% – ouch;

and

> Granularity of the volume control is very limited - my DAC has 64 vol positions but spotifyd doesn't seem to be that fine-grained;

When you change the volume through the Spotify app, you are using a software volume control. This degrades the sound quality, so you'd probably be better off changing the volume elsewhere anyway. Either in the receiver/amplifier or whatever you are connecting the DAC to, or the DAC itself using something like alsamixer.

Of course, it could be nice to have the volume stored anyway 😃

> When I play an album it seems to play one song and then stop

Yes, I noticed this too. I will investigate. I think it didn't happen on x86, hence I didn't catch it (I developed on the desktop, and only gave RPI a quick try).

> Closing the Spotify app on iPhone and then restarting the app doesn't seem to detect that music is playing on spotfiyd

This sounds like a bug in librespot. The author says that the Spotify connect protocol is not correct, hence the rewrite to improve it.

> Verbose output is too verbose – I tried to take look at the verbose logs but there was just so much output!

What I do is to filter out the messages I don't want. That is, run something like this:
```
spotifyd --no-daemon -v | grep -v INFO
```
Which gives more reasonable output.

**Comment by SimonPersson** [comment]
I tried building natively on a RPi today, and can confirm that skipping around in songs works, and it does keep playing after the first song finishes.

It seems like something goes wrong during the compilation of libvorbis. I will see what I can do about that.

**Comment by SimonPersson** [comment]
I've fixed the libvorbis issue by switching to tremor as a vorbis decoder instead. Could you verify that skipping back and forth, as well as continuing with next song in an album works for you too? If so, I will close this issue for now, as the other stuff depends on librespot, and I will look at it once the rewrite of librespot is in place.

**Comment by wedi** [comment]
I tried the binary build from 27-03-2017 and it works like a charm! 😃👍🏻🎉

Skipping songs works and playing continues after the first song in an album.
For it remembers the volume as well. Maybe I am using the alsamixer already? This thing kept being a secret for me while messing with https://github.com/Fornoth/spotify-connect-web.
